samza-core/src/main/scala/org/apache/samza/util/Util.scala

```diff
@@ -62,6 +62,21 @@ object Util extends Logging {
     }
   }
 
+  /**
+    * Instantiate an object from given className, and given constructor parameters.
+    */
+  def getObj(className: String, constructorParameterTypes: (Class[_], Object)*) = {
```

Comment:
As a general util method, I think it makes more sense for this to throw instead for returning an Option. You can then catch at your call site. 
Should also add explicit return type for public methods.
Maybe s/contructorParamaterTypes/constructorParams

Reply:
Also, I think as written the return type will be Any (== Object) since in the happy path it returns an instance?

Reply:
I was trying to avoid the nested try-catch. 
fixed.